### PR TITLE
Chapter 2 - The Observer Pattern

### PR DESCRIPTION
**Pull request by a-panayotopoulos**
A simple RSS-like system showing subscriptions to different sources.

### CODE REVIEW

File: axp-test/org/axp/observer/ObserverTest.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+				+ "et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut "
+				+ "aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum "
+				+ "dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui "
+				+ "officia deserunt mollit anim id est laborum.", "http://axpblog/152-reasons-I-l" ),
+		new BlogPost( "Alex P", "38 hilarious cat GIFs",
+				"In est metus, imperdiet a finibus vel, blandit nec mi. Morbi id augue vitae leo rutrum tempus. Nullam "
+				+ "ullamcorper nisl dui, non posuere nisi porta et. Interdum et malesuada fames ac ante ipsum primis in "
+				+ "faucibus. Phasellus accumsan lacinia urna, imperdiet placerat nulla mollis at. Vivamus ac mauris velit. "
+				+ "Mauris varius vitae massa in tincidunt. Nulla facilisis pellentesque finibus. Praesent lobortis auctor "
+				+ "urna ut sodales.", "http://axpblog/38-hilarious-ca" ),
```
Comment (clearscene) [inline axp-test/org/axp/observer/ObserverTest.java]: Support: Where are my cat gifs? This site must be a scam!

Comment (a-panayotopoulos) [inline axp-test/org/axp/observer/ObserverTest.java]: http://gifrific.com/wp-content/uploads/2014/01/Cat-Closes-Box-on-Cat-and-Sits-on-It.gif

File: axp/org/axp/observer/display/AbstractDisplay.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		this.out = out;
+	}
+
+	public void update( Observable o, Object arg ) {
+		if ( arg == null ) {
+			throw new IllegalArgumentException( "Got a null Post" );
+		}
+		if ( !( arg instanceof Post ) ) {
+			throw new IllegalArgumentException( "Can only display Post objects, not " + arg.getClass().getName() );
+		}
```
Comment (Utopia1138) [inline axp/org/axp/observer/display/AbstractDisplay.java]: It's a shame `java.util.Observer` (presumably?) predated generics. This should be enforced by the type system.

Comment (a-panayotopoulos) [inline axp/org/axp/observer/display/AbstractDisplay.java]: Very much so.  :crying_cat_face:

**Pull request merged**